tests/unittests/sources/test_wsl.py

```diff
@@ -460,6 +461,53 @@ def test_get_data_jinja(self, m_lsb_release, paths, tmpdir):
             cast(MIMEMultipart, ud), "text/cloud-config"
         ), "No cloud-config part should exist"
 
+    @pytest.mark.parametrize("with_agent_data", [True, False])
+    @mock.patch("cloudinit.util.lsb_release")
+    def test_get_data_x(
+        self, m_lsb_release, with_agent_data, caplog, paths, tmpdir
+    ):
+        """
+        Assert behavior of empty .cloud-config dir with and without agent data
+        """
+        m_lsb_release.return_value = SAMPLE_LINUX_DISTRO
+        data_path = tmpdir.join(".cloud-init", f"{INSTANCE_NAME}.user-data")
+        data_path.dirpath().mkdir()
+
+        if with_agent_data:
+            ubuntu_pro_tmp = tmpdir.join(".ubuntupro", ".cloud-init")
+            os.makedirs(ubuntu_pro_tmp, exist_ok=True)
+            agent_path = ubuntu_pro_tmp.join("agent.yaml")
+            agent_path.write(
+                "#cloud-config\nlandscape:\n client: {account_name: agent}\n"
+            )
+
+        ds = wsl.DataSourceWSL(
+            sys_cfg=SAMPLE_CFG,
+            distro=_get_distro("ubuntu"),
+            paths=paths,
+        )
+
+        assert ds.get_data() is with_agent_data
+        ud = ds.get_userdata(True)
+        print(ud)
+
+        expected_log_level = logging.INFO if with_agent_data else logging.ERROR
+        regex = (
+            "Unable to load any user-data file in /[^:]*/.cloud-init:"
+            " /.*/.cloud-init directory is empty"
+        )
+        messages = [
+            x.message
+            for x in caplog.records
+            if x.levelno == expected_log_level and re.match(regex, x.message)
+        ]
+        assert (
+            len(messages) > 0
+        ), "Expected log message matching '{}' with log level '{}'".format(
+            regex, expected_log_level
+        )
+        assert ud is not None
```

Comment:
This only asserts that get_userdata() returns a MIME message (which passes for both absent and present userdata). Let's actually assert that correct ds.userdata_raw or None was found for each case instead.